src/main/java/net/minecraftforge/common/tags/ForgeTagBuilder.java

```diff
@@ -0,0 +1,281 @@
+package net.minecraftforge.common.tags;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSortedSet;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.tags.Tag;
+import net.minecraft.tags.Tag.ITagEntry;
+import net.minecraft.util.JsonUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+public final class ForgeTagBuilder<T> extends Tag.Builder<T>
+{
+    /**
+     *
+     * @param <T> The type of {@link IForgeRegistryEntry}. Completely irrelevant for the Comparator. Just here to save People using it from unnecessary casting.
+     * @return A comparator comparing {@link IForgeRegistryEntry} by their registryName's naturalOrder ({@link ResourceLocation} comparison).
+     */
+    public static <T> Comparator<IForgeRegistryEntry<T>> registryNameComparator()
+    {
+        return Comparator.comparing(IForgeRegistryEntry::getRegistryName, Comparator.naturalOrder());
```

Comment:
`Comparator.comparing(IForgeRegistryEntry::getRegistryName)` should be sufficient.

Reply:
Oh, didn't know this works... Thanks...